### PR TITLE
docs: misc fixes

### PR DESCRIPTION
**Pull request by misterjones**
## Changes

- Hyphenate "higher order" when referring to "higher-order components."
- Remove apostrophe from pluralized form of HOC (HOCs), as it's unnecessary.
- Change instances of "hoc" to "HOC".
- Fix miscellaneous code comments.
  - Fix typos, formatting, etc. in code comments of miscellaneous files.
- Fix a few broken/empty links.
- Clean up "Tokens and Higher-Order Components in TypeScript" page.
  - Fix typos, formatting, etc.

### CODE REVIEW

File: packages/bodiless-layouts/src/meta/index.tsx [hunk 1]
```diff
@@ -104,16 +104,16 @@ const withTerm = (cat: string) => (term: string):HOC => Component => (
...
- * @param hoc the hoc to wrap.
+ * @param hoc the HOC to wrap.
  */
 const perserveMeta = (hoc: HOC): HOC => Component => (
   withMeta(Component)(hoc(Component) as ComponentType<any>)
 );
 
 /**
- * withFacet is expect to be passed to an on function and takes a term and and hoc (using curring)
+ * withFacet is expect to be passed to an on function and takes a term and and HOC (using curring)
```
Comment (KaterinaPashkevich) [inline packages/bodiless-layouts/src/meta/index.tsx:116]: please fix the typo in the row 116
```suggestion
 * withFacet is expect to be passed to an on function and takes a term and a HOC (using curring)
```

File: packages/bodiless-layouts/src/meta/index.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ function customizer(objValue:any, srcValue:any) {
 }
 const asPassThough = (Component:CTWM) => Component;
 /**
- * Creates an HOC use it to attach meta data in an hoc.
+ * Creates an HOC use it to attach meta data in an HOC.
```
Comment (KaterinaPashkevich) [inline packages/bodiless-layouts/src/meta/index.tsx:32]:
```suggestion
 * Creates a HOC use it to attach meta data in a HOC.
```

File: packages/bodiless-layouts/src/meta/index.tsx [hunk 3]
```diff
@@ -104,16 +104,16 @@ const withTerm = (cat: string) => (term: string):HOC => Component => (
   withMeta({ categories: { [cat]: [term] } })(Component)
 );
 /**
- * preserveMeta returns takes an hoc and returns another one that will apply the hoc but preserve
+ * preserveMeta returns takes an HOC and returns another one that will apply the HOC but preserve
```
Comment (KaterinaPashkevich) [inline packages/bodiless-layouts/src/meta/index.tsx:107]:
```suggestion
 * preserveMeta returns takes a HOC and returns another one that will apply the HOC but preserve
```

File: packages/bodiless-tokens/README.md [hunk 4]
```diff
@@ -76,7 +76,7 @@ const withComposedToken = flowHoc(
 However, there are a few key differences:
 
 - Metadata (static properties) attached to a component are prppagated through
```
Comment (KaterinaPashkevich) [inline packages/bodiless-tokens/README.md:78]:
```suggestion
- Metadata (static properties) attached to a component are propagated through
```

File: packages/fclasses/src/Design.tsx [hunk 5]
```diff
@@ -193,13 +193,13 @@ export const extendDesignable = (transformDesign: TransformDesign = identity) =>
...
+ * HOCs to apply to each sub-component via the `withDesign` HOC.
  *
  * @param startComponents
  * An object defining the set of constituent subcomponents. Each key
  *   is a string which identifies the component. Each value is the component itself, which
- *   will be modified by any HOC's provided by withDesign.
+ *   will be modified by any HOCs provided by withDesign.
  *
  * @return
  * An HOC which yields a designable version of the component to which it is applied.
```
Comment (KaterinaPashkevich) [inline packages/fclasses/src/Design.tsx:205]:
```suggestion
 * A HOC which yields a designable version of the component to which it is applied.
```

File: packages/fclasses/src/tokenSpec.ts [hunk 6]
```diff
@@ -109,7 +109,7 @@ function as(
  * The design to apply.
  *
  * @returns
- * An hoc which applies the design.
+ * An HOC which applies the design.
```
Comment (KaterinaPashkevich) [inline packages/fclasses/src/tokenSpec.ts:112]:
```suggestion
 * A HOC which applies the design.
```

File: packages/fclasses/src/tokenSpec.ts [hunk 7]
```diff
@@ -185,7 +185,7 @@ const extendDomain = extendDesign;
  * list will be converted to designs by invoking them on an empty design.
  *
  * @param dx
- * List of designs or HODs which will be used as extensons
+ * List of designs or HODs which will be used as extensions.
  *
  * @returns
  * An HOD which will extend the base design with the supplied designs.
```
Comment (KaterinaPashkevich) [inline packages/fclasses/src/tokenSpec.ts:191]:
```suggestion
 * A HOD which will extend the base design with the supplied designs.
```

File: packages/fclasses/src/types.ts [hunk 8]
```diff
@@ -307,30 +307,30 @@ export type ReservedDomains<
...
   Meta?: TokenMeta,
 };
 
 /**
  * Type of an argument to `as` and/or the value of a key in an Extended Design.
  * May be:
- * - A token specified in token object notation
- * - A token specified as an HOC
+ * - A token specified in token object notation;
+ * - A token specified as an HOC;
```
Comment (KaterinaPashkevich) [inline packages/fclasses/src/types.ts:333]:
```suggestion
 * - A token specified as a HOC;
```

File: packages/fclasses/src/withHocDesign.tsx [hunk 9]
```diff
@@ -23,10 +23,10 @@ import { flowHoc } from './flowHoc';
  * @hidden
  * Creates an HOC which applies a specified design to the wrapped component.
```
Comment (KaterinaPashkevich) [inline packages/fclasses/src/withHocDesign.tsx:24]:
```suggestion
 * Creates a HOC which applies a specified design to the wrapped component.
```

File: packages/bodiless-layouts/src/meta/index.tsx (hunk 1 above)
Comment (misterjones) [reply packages/bodiless-layouts/src/meta/index.tsx]: ✅ Fixed.

File: packages/bodiless-layouts/src/meta/index.tsx (hunk 2 above)
Comment (misterjones) [reply packages/bodiless-layouts/src/meta/index.tsx]: ✅ Fixed.

File: packages/bodiless-layouts/src/meta/index.tsx (hunk 3 above)
Comment (misterjones) [reply packages/bodiless-layouts/src/meta/index.tsx]: ✅ Fixed.

File: packages/bodiless-tokens/README.md (hunk 4 above)
Comment (misterjones) [reply packages/bodiless-tokens/README.md]: ✅ Fixed.

File: packages/bodiless-tokens/README.md [hunk 10]
```diff
@@ -76,7 +76,7 @@ const withComposedToken = flowHoc(
...
-  the chain of HOC's.
+  the chain of HOCs.
 - If you are using Typescript, the type of the parameters is constrained to be an
```
[the remark this answers is not in the document]
Comment (misterjones) [reply packages/bodiless-tokens/README.md]: ✅ Fixed.

File: packages/fclasses/README.md [hunk 11]
```diff
@@ -82,7 +82,7 @@ const withComposedToken = flowHoc(
 However, there are a few key differences:
 
 - Metadata (static properties) attached to a component are prppagated through
```
[the remark this answers is not in the document]
Comment (misterjones) [reply packages/fclasses/README.md]: ✅ Fixed.

File: packages/fclasses/README.md [hunk 12]
```diff
@@ -82,7 +82,7 @@ const withComposedToken = flowHoc(
...
-  the chain of HOC's.
+  the chain of HOCs.
 - If you are using Typescript, the type of the parameters is constrained to be an
```
[the remark this answers is not in the document]
Comment (misterjones) [reply packages/fclasses/README.md]: ✅ Fixed.

File: packages/fclasses/src/Design.tsx (hunk 5 above)
Comment (misterjones) [reply packages/fclasses/src/Design.tsx]: ✅ Fixed.

File: packages/fclasses/src/tokenSpec.ts (hunk 6 above)
Comment (misterjones) [reply packages/fclasses/src/tokenSpec.ts]: ✅ Fixed.

File: packages/fclasses/src/tokenSpec.ts (hunk 7 above)
Comment (misterjones) [reply packages/fclasses/src/tokenSpec.ts]: ✅ Fixed.

File: packages/fclasses/src/tokenSpec.ts [hunk 13]
```diff
@@ -248,13 +248,13 @@ const on = (
...
  * specification created by merging those partials.  Each parameter may be:
  * - a partial tokens specification object, in which case the keys must belong to the set of
- *   allowed domains.  Keys will be re-ordered to match the canonoical order, and missing keys
+ *   allowed domains.  Keys will be re-ordered to match the canonical order, and missing keys
  *   will be supplied.
  * - a string, which will be treated as a set of classes to be applied to the `_` key of the
- *   `Core` domain.  The `_` key in a design applies HOC's or classes to the component as a
+ *   `Core` domain.  The `_` key in a design applies HOCs or classes to the component as a
  *   whole rather than one of its slots.
  * - a function, which will be treated as an HOC to be applied to the `_` key of the `Core`
```
[the remark this answers is not in the document]
Comment (misterjones) [reply packages/fclasses/src/tokenSpec.ts]: ✅ Fixed.

File: packages/fclasses/src/types.ts (hunk 8 above)
Comment (misterjones) [reply packages/fclasses/src/types.ts]: ✅ Fixed.

File: packages/fclasses/src/withHocDesign.tsx (hunk 9 above)
Comment (misterjones) [reply packages/fclasses/src/withHocDesign.tsx]: ✅ Fixed.

**Comment by misterjones** [comment]
Hi, @KaterinaPashkevich.
Thank you for catching the typos. 😄
Feedback addressed. 👍

I updated all instances of "an HOC" to "a HOC" in the codebase via two a two additional commits — one updating all the instances in markdown files ([f1e7b5f](https://github.com/johnsonandjohnson/Bodiless-JS/pull/2138/commits/f1e7b5f446623312ba462244a5ded2826bc43fab)) and the other updating all the instances in code comments ([aa65cc0](https://github.com/johnsonandjohnson/Bodiless-JS/pull/2138/commits/aa65cc0aa37f1ba470610e38cefa076c7ff05de4)).

Given the number of changed files, I _only_ updated these instances, and ignored all other typos.

**Review by hvanyo — APPROVED** [review]

**Review by KaterinaPashkevich — APPROVED** [review]

**Pull request merged**